json_serializable/lib/src/type_helpers/json_converter_helper.dart

```diff
@@ -49,6 +62,19 @@ class JsonConverterHelper extends TypeHelper {
 
     final asContent = asStatement(converter.jsonType);
 
+    if (!converter.jsonType.isNullableType && targetType.isNullableType) {
+      // Hacky way to hoist the expression to avoid casting
+      // Ideally we should be able to cleanly declare a variable in the
+      // parent code-block, instead of using (){}()
+      return '''
+() {
```

Comment:
Can you "DRY" up this expression with the one above? Put it in a private, static helper?